src/scripts/cfo.zig

```diff
@@ -0,0 +1,484 @@
+
+const stdx = @import("../stdx.zig");
+const flags = @import("../flags.zig");
+const fatal = flags.fatal;
+const Shell = @import("../shell.zig");
+
+pub const CliArgs = struct {
+    budget_minutes: u64 = 10,
+    hang_minutes: u64 = 30,
+    concurrency: u32 = 16,
+};
+
+const Fuzzer = enum {
+    ewah,
+    lsm_cache_map,
+};
+
+pub fn main(shell: *Shell, gpa: std.mem.Allocator, cli_args: CliArgs) !void {
+    if (builtin.os.tag == .windows) {
+        return error.NotSupported;
+    }
+
+    _ = gpa;
+    assert(try shell.exec_status_ok("git --version", .{}));
+
+    const commit_sha_str = try shell.exec_stdout("git rev-parse HEAD", .{});
+    assert(commit_sha_str.len == 40);
+    const commit_sha: [40]u8 = commit_sha_str[0..40].*;
+
+    const commit_timestamp_str = try shell.exec_stdout(
+        "git show -s --format=%ct {sha}",
+        .{ .sha = commit_sha_str },
+    );
+    const commit_timestamp = try std.fmt.parseInt(u64, commit_timestamp_str, 10);
+
+    log.info("commit={s} timestamp={s}", .{ commit_sha_str, commit_timestamp_str });
+
+    const random = std.crypto.random;
+
+    try shell.exec("zig/zig build -Drelease build_fuzz", .{});
+
+    const FuzzerChild = struct {
+        child: std.ChildProcess,
+        seed: SeedRecord,
+    };
+
+    var seeds = std.ArrayList(SeedRecord).init(shell.arena.allocator());
+    var fuzzers = try shell.arena.allocator().alloc(?FuzzerChild, cli_args.concurrency);
+    for (fuzzers) |*fuzzer| fuzzer.* = null;
+    defer for (fuzzers) |*fuzzer_or_null| {
+        if (fuzzer_or_null.*) |*fuzzer| {
+            _ = fuzzer.child.kill() catch {};
+            fuzzer_or_null.* = null;
+        }
+    };
+
+    const total_budget_seconds =
+        (cli_args.budget_minutes + cli_args.hang_minutes) * std.time.s_per_min;
+    for (0..total_budget_seconds) |second| {
+        const last_iteration = second == total_budget_seconds - 1;
+
+        if (second < cli_args.budget_minutes * std.time.s_per_min) {
+            // Start new fuzzer processes if we have more time.
+            for (fuzzers) |*fuzzer_or_null| {
+                if (fuzzer_or_null.* == null) {
+                    const fuzzer = random.enumValue(Fuzzer);
+                    const seed = random.int(u64);
+                    const command = try shell.print(
+                        "zig/zig build -Drelease fuzz -- --seed={d} {s}",
+                        .{ seed, @tagName(fuzzer) },
+                    );
+                    const seed_record = SeedRecord{
+                        .commit_timestamp = commit_timestamp,
+                        .commit_sha = commit_sha,
+                        .fuzzer = fuzzer,
+                        .ok = false,
+                        .seed_timestamp = @as(u64, @intCast(std.time.timestamp())),
+                        .seed = seed,
+                        .command = command,
+                    };
+                    log.debug("will start '{s}'", .{command});
```

Comment:
Maybe cleanup of these log messages (the corresponding `will reap` one below is more confusing, it makes it sound like something odd is happening when its just the fuzzer being done)

Reply:
> comes from how VS Code

No wonder I found it confusing! :laughing: 

(It might be from how I normally use  `will` when I write my log comments)